packages/core/test/http/requestBuilder.test.ts

```diff
@@ -562,6 +562,43 @@ describe('test default request builder behavior with succesful responses', () =>
     );
     expect(optionalString.result).toEqual(undefined);
   });
+
+  it('should test request builder query parameters with number, string, bool and BigInt', async () => {
+    const expectedRequestUrl =
+      'http://apimatic.hopto.org:3000/test/requestBuilder?number=12345&string=string&bool=true&biginit=12345';
+
+    const reqBuilder = defaultRequestBuilder('GET');
+    reqBuilder.appendPath('/test/requestBuilder');
+    reqBuilder.baseUrl('default');
+    reqBuilder.query('number', 12345);
+    reqBuilder.query('string', 'string');
+    reqBuilder.query('bool', true);
+    reqBuilder.query('biginit', BigInt(12345));
+
+    const apiResponse = await reqBuilder.callAsText();
+    expect(apiResponse.request.url).toEqual(expectedRequestUrl);
+  });
+
+  it('should test request builder query parameters with complex object', async () => {
```

Comment:
Please add another test to check the functionality of sending array of native types as query parameters

Reply:
test added for native array with supported serialization formats.